sklearn/impute/_iterative.py

```diff
@@ -581,6 +580,37 @@ def fit_transform(self, X, y=None):
             self.n_iter_ = 0
             return super()._concatenate_indicator(Xt, X_indicator)
 
+        # If min or max value are (scalar) None, convert them to - or + inf
```

Comment:
I agree with Joel. Also remove all comments. The code is self-explanatory. We try to limit comment for aspect which is not trivial by just reading the code. If the operation is not obvious, it might mean that we need to change the code then :)

Reply:
To be more explicit, I think that `_validate_limit` should only validate a single limit. We have the exact same sequence of operation for min and max. The differences should be triggered by the parameter passed to this method.

Reply:
So the function should be:

```python
@staticmethod
    def _validate_limit(limit, limit_type, n_features):
        limit_bound = np.inf if limit_type == "max" else -np.inf
        limit = limit_bound if limit is None else limit
        if np.isscalar(limit):
            limit = np.full(n_features, limit)
        limit = check_array(
            limit, force_all_finite=False, copy=False, ensure_2d=False
        )
        if not limit.shape[0] == n_features:
            raise ValueError(
            f"'{limit_type}_value' should be of shape ({n_features},) when "
            f"an array-like is provided. Got {limit.shape}, instead."
        )
        return limit
```

Reply:
Done - thank you for fleshing out the method :)